dialogue-serde/src/test/java/com/palantir/conjure/java/dialogue/serde/ConjureBodySerDeTest.java

```diff
@@ -137,14 +141,12 @@ public void testRequestUnknownContentType() throws IOException {
     }
 
     @Test
-    public void testErrorsDecoded() {
-        TestResponse response = new TestResponse().code(400);
-
+    public void testErrorsDecoded() throws JsonProcessingException {
         ServiceException serviceException = new ServiceException(ErrorType.INVALID_ARGUMENT);
-        SerializableError serialized = SerializableError.forException(serviceException);
-        errorDecoder = mock(ErrorDecoder.class);
-        when(errorDecoder.isError(response)).thenReturn(true);
-        when(errorDecoder.decode(response)).thenReturn(new RemoteException(serialized, 400));
+        TestResponse response = TestResponse.withBody(
```

Comment:
This is to have an "e2e" test where the error decoder is actually called (instead of mocked) - to validate that replacing the `EncodingDeserializerRegistry` with `EncodingDeserializerForEndpointRegistry` doesn't introduce a regression.

I will also port over the tests from `ErrorDecoderTest` to test the new `EndpointErrorDecoder`